entities/weapons/weapon_zs_fastzombie_hunter/shared.lua

```diff
@@ -0,0 +1,387 @@
+end

+

+function SWEP:StopMoaning()
+end

+

+function SWEP:StartMoaningSound()
+end

+

+function SWEP:PlayHitSound()
+	self.Owner:EmitSound("NPC_FastZombie.AttackHit")
+end

+

+function SWEP:PlayMissSound()
+	self.Owner:EmitSound("NPC_FastZombie.AttackMiss")
+end

+

+function SWEP:PlayAttackSound()
+end

+

+function SWEP:PlayIdleSound()
+	self.Owner:EmitSound("NPC_FastZombie.AlertFar")
+	self:SetRoarEndTime(CurTime() + self.RoarTime)
+end

+

+function SWEP:PlayAlertSound()
+	self.Owner:EmitSound("NPC_FastZombie.Frenzy")
+	self:SetRoarEndTime(CurTime() + self.RoarTime)
+end

+

+function SWEP:StartSwingingSound()
+	self.Owner:EmitSound("NPC_FastZombie.Gurgle")
+end

+function SWEP:StopSwingingSound()
+	self.Owner:StopSound("NPC_FastZombie.Gurgle")
+end

+function SWEP:IsMoaning()
+	return false
+end

+

+function SWEP:Move(mv)
+	if self:IsPouncing() or self:GetPounceTime() > 0 then
+		mv:SetMaxSpeed(0)
+		mv:SetMaxClientSpeed(0)
+	elseif self:GetClimbing() then
+		mv:SetMaxSpeed(0)
+		mv:SetMaxClientSpeed(0)
+
+		local owner = self.Owner
+		local tr = self:GetClimbSurface()
+		local angs = self.Owner:SyncAngles()
+		local dir = tr and tr.Hit and (tr.HitNormal.z <= -0.5 and (angs:Forward() * -1) or math.abs(tr.HitNormal.z) < 0.75 and tr.HitNormal:Angle():Up()) or Vector(0, 0, 1)
+		local vel = Vector(0, 0, 4)

+		if owner:KeyDown(IN_FORWARD) then
+			vel = vel + dir * 160
+		end
+		if owner:KeyDown(IN_BACK) then
+			vel = vel + dir * -160
+		end

+		if vel.z == 4 then
+			if owner:KeyDown(IN_MOVERIGHT) then
+				vel = vel + angs:Right() * 60
+			end
+			if owner:KeyDown(IN_MOVELEFT) then
+				vel = vel + angs:Right() * -60
+			end
+		end

+		mv:SetVelocity(vel)

+

+		return true
+	elseif self:GetSwinging() then
+		mv:SetMaxSpeed(math.min(mv:GetMaxSpeed(), 60))
+		mv:SetMaxClientSpeed(math.min(mv:GetMaxClientSpeed(), 60))
+	end
+end

+

+function SWEP:SetRoarEndTime(time)
+	self:SetDTFloat(1, time)
+end

+

+function SWEP:GetRoarEndTime()
+	return self:GetDTFloat(1)
+end

+

+function SWEP:IsRoaring()
+	return CurTime() < self:GetRoarEndTime()
+end

+

+function SWEP:SetPounceTime(time)
+	self:SetDTFloat(2, time)
+end

+

+function SWEP:GetPounceTime()
+	return self:GetDTFloat(2)
+end

+

+function SWEP:SetPounceTime(time)
+	self:SetDTFloat(2, time)
+end

+

+function SWEP:GetPounceTime()
+	return self:GetDTFloat(2)
+end

+

+function SWEP:SetClimbing(climbing)
+	self:SetDTBool(1, climbing)
+end

+

+function SWEP:GetClimbing()
+	return self:GetDTBool(1)
+end

+SWEP.IsClimbing = SWEP.GetClimbing

+function SWEP:SetSwinging(swinging)
+	self:SetDTBool(2, swinging)
+end

+

+function SWEP:GetSwinging()
+	return self:GetDTBool(2)
+end

+

+function SWEP:SetPouncing(leaping)
+	self:SetDTBool(3, leaping)
+end

+

+function SWEP:GetPouncing()
+	return self:GetDTBool(3)
+end

+SWEP.IsPouncing = SWEP.GetPouncing

+

+if CLIENT then return end

+

+function SWEP:Deploy()
+	self.Owner:CreateAmbience("fastzombieambience")


```

Comment:
where is the "fastzombieambience" ambience? Also please fix the linings so there is no "^M"

Reply:
fastzombieambience already in gamemode https://github.com/MrCraigTunstall/zombiesurvival/tree/master/entities/entities/status_fastzombieambience